modules/bullet/lib/bullet/impl.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Bullet.Impl do
+
+  defp distance(x1, x2, y1, y2) do
+    dx = x1-x2
+    dy = y1-y2
+    (dx*dx) + (dy*dy)
+  end
+
+  defp did_collide?(x1, x2, y1, y2) do
+    rad = Application.get_env(:player, :radius) + Application.get_env(:bullet, :radius)
+    distance(x1, x2, y1, y2) < rad*rad
+  end
+
+  defp inner_filter(x1, y1, %{x: x2, y: y2}) do
+    did_collide?(x1, x2, y1, y2)
+  end
+  defp find_collisions(x1, y1, pid) do
+    inner_filter(x1, y1, GenServer.call(pid, {:peek}))
+  end
+
+  def calculate_collisions(%{x: x1, y: y1 }, players) do
```

Comment:
Rather than checking for Process.alive, I wonder if you could have monitored the processes and just updated the list on exit?

No deduction, just a thought...